SCYLLA-VERSION-GEN

```diff
@@ -88,10 +88,11 @@ else
 	SCYLLA_VERSION=$VERSION
 	if [ -z "$SCYLLA_RELEASE" ]; then
 		GIT_COMMIT=$(git -C "$SCRIPT_DIR" log --pretty=format:'%h' -n 1 --abbrev=12)
-		# For custom package builds, replace "0" with "counter.your_name",
+		# For custom package builds, replace "0" with "counter.yourname",
 		# where counter starts at 1 and increments for successive versions.
 		# This ensures that the package manager will select your custom
 		# package over the standard release.
+		# Do not use any special characters like - or _ in the name above!
```

Comment:
could add something like
```sh
if echo "$SCYLLA_BUILD" | grep -E -q "_|-"; then
  echo "Do not use any special characters like - or _ in SCYLLA_BUILD!" 1>&2
  exit 1
fi
```
to be more foolproof. or better of checking `$SCYLLA_RELEASE` instead, and moving this check down right before we write it to `"$OUTPUT_DIR/SCYLLA-VERSION-FILE"`. as we also have customize point for `SCYLLA_RELEASE`. it allows one to override it using a `version` file.

also, i could guess that `-` is used as the delimiter of `$product-$version-$release`. but why `_` is forbidden? is it used for some other purpose?

Reply:
If `_` is used, the debian package builder will fail, with an error that `_` is illegal character in the version (or that part of the version).